content_sync/pipelines/definitions/concourse/remove_unpublished_sites.py

```diff
@@ -212,6 +218,22 @@ def __init__(self, **kwargs):
             ),
         ]
         job = Job(name=self._remove_unpublished_sites_job_identifier, serial=True)
-
+        if not is_dev():
+            resource_types.append(slack_notification_resource())
+            resources.append(self._slack_resource)
+            job.on_failure = SlackAlertStep(
+                alert_type="failed",
+                text=f"""
+                Failed to remove unpublished site {short_id} (Site Name: {site_name}). Check the pipeline logs for more details.
```

Comment:
> The error messages will not be able to run in development. I added site_name, and short_id, which will be used via Vault in RC/Production, but they won't work locally.

I get what you're going for here, but I don't think that this works exactly like you're intending it to. `short_id` and `site_name` are not set via Vault in RC / production. The `get_unpublished_sites_task` fetches the list of sites to be unpublished from `ocw-studio` then iterates them in an `AcrossStep`:

```python
            DoStep(
                do=across_tasks,
                across=[
                    AcrossVar(
                        var="site",
                        values=f"((.:{self._unpublished_sites_var_identifier}.sites))",
                        max_in_flight=5,
                    )
                ],
            ),
```

This creates the `site` namespace, but it is only available from within tasks running in the `AcrossStep`. When the pipeline fails *outside* the across step, such as when hitting the `ocw-studio` API to get the list of sites, `site_name` and `short_id` are not available. If we want to include this information in the Slack alert, we would need to provide different `on_failure` / `on_abort` steps both inside and outside the context of the `AcrossStep`. The best way to do this would be to wrap the tasks in `across_tasks` in `add_error_handling` or manually setting the `on_failure` / `on_abort` of these steps. The global error handler on the job should not try to use these variables, as they are not available outside the context of the `AcrossStep`. It might be best just to remove this information from the error entirely. I've found that when debugging an issue it isn't super helpful to have this information in Slack, and I think this is especially true with the unpublish sites pipeline, since there is only one pipeline with no instance vars. The purpose of the Slack alert is to let you know that something went wrong. Generally you are just going to click the link to bring up the pipeline, and detailed error information will be available in the pipeline logs.

Reply:
Thanks for the detailed explanation 👍🏻 I'll update the PR to remove the step. The error alert (with the link) itself should be sufficient.